schema/2023-03/Form.json

```diff
@@ -230,6 +230,13 @@
         "isAnswerHashed": {
           "description": "Boolean specifying if the answer should be hashed or not.",
           "type": "boolean"
+        },
+        "answerOptionsVisibilityFilter": {
+          "description": "List of answerOptions that will proc the visibility of this element",
+          "type": "array",
+          "items": {
+            "$ref": "#/$defs/answerOption"
```

Comment:
Would this be a copy of the whole answerOption? If yes, maybe its ID is enough. 

Do we also need the element-ID (+ subElement ID, if matrix)?

Reply:
@janmlam 

Can the answerOptionsVisibilityFilter relate to the answeroptions of multiple elements? Or is alwas just one element?

Suggestion for _always depended on only one element_:

```json

{
  "answerOptionsVisibilityFilter": {
    "externalElementId": "var1",
    "externalAnswerOptionIds": [
      "option1", 
      "option3"
    ]
  }
}
```

Suggestion for _depended on multiple one element_:

```json

{
  "answerOptionsVisibilityFilter": [
    {
       "externalElementId": "var1",
       "externalAnswerOptionId": "option1"
    },
    {
       "externalElementId": "var2",
       "externalAnswerOptionId": "some_other_option2"
    }
  ]
}
```

Reply:
@f-krull @janmlam  Your first suggestion should be sufficient. It's only possible to select one dependency pr element as this is a dropdown in the GUI. If an element depended on multiple elements to be shown it would have to have a rule if the first element says show and the second says don't show.

Reply:
We have a PR that changes to use external Ids, it currently look like this if an element is set to show if cb_choice_1 or cb_choice_3 is selected of another element with external_id value cb_ele
![image](https://github.com/unioslo/survey-api-schemas/assets/54781898/a7e59aa3-7dbd-431e-8a1f-9a17af75bb3b)
Ill update this pr when our other pr is merged

Reply:
@janmlam  Can your explain the reasoning for the JSON structure in your screenshot? 
Why is `externalAnswerOptionIds` an array when "cb_ele" shows up twice?

how about something like this?: 

```json
{
  "answerOptionsVisibilityFilter": {
    "elements": {
       "cb_ele": {
         "answerOptionIds": [
           "cb_choice_1", "cb_choice_3"
         ]
       },
       "cb_ele2": {
         "externalAnswerOptionIds": [
           "cb_choice_xy"
         ]
       }
  }
}
```
This will prevent "cb_ele" being used twice.

Reply:
@kajh commented on this as well https://github.uio.no/IT-NETTSKJEMA/nettskjema-api/pull/3460#discussion_r21348 is it alright to take the discussion there?